Source/WebCore/rendering/RenderBlock.cpp

```diff
@@ -837,14 +837,34 @@ void RenderBlock::setLogicalTopForChild(RenderBox& child, LayoutUnit logicalTop,
     }
 }
 
+bool RenderBlock::childIsFloatWithTrimmedMargin(const RenderBox& child) const
+{
+    if (!child.isFloating() || child.needsLayout())
+        return false;
+
+    // A child's margin should be considered trimmed if:
+    // 1.) The computed value for the margin in its m_marginBox is 0
+    // 2.) The respective margin's value in its style is auto OR
+    //     the computed value for the margin comes out to something other than 0
```

Comment:
it should simply read along the lines of "used value != computed value and used value is zero"

Reply:
maybe the code below should reflect that